### PR TITLE
Apmahade/externalstoragereads

### PR DESCRIPTION
**Pull request by appym31**
Currently in getRef, scribe sends a get request to historian which goes to gitrest, and finally to external storage, so that we read from external storage always. 

However, we only want to read and write to external storage if a tenant is linked to external storage.

In the write flow, scribe sends a config value which indicates whether or not to write to external storage. In historian, we check tenant details to see if external storage is linked. If it is not linked, we set the config value to false. Gitrest will read the false value, and will not write to external storage. 

This PR adds the necessary interface so scribe can send a config as part of the request for the read flow as well, similar to write.

In a later PR, we will modify historian to set the value according to the tenant details, and modify gitrest to read from external storage if the value is true.

### CODE REVIEW

File: server/routerlicious/packages/services-client/src/gitManager.ts [hunk 1]
```diff
@@ -8,7 +8,10 @@ import * as resources from "@fluidframework/gitresources";
 import { buildHierarchy } from "@fluidframework/protocol-base";
 import * as api from "@fluidframework/protocol-definitions";
 import { debug } from "./debug";
-import { ICreateRefParamsExternal, IPatchRefParamsExternal, IGitManager, IHistorian } from "./storage";
```
Comment (tanviraumi) [inline server/routerlicious/packages/services-client/src/gitManager.ts:11]: Why changing this line?

**Review by tanviraumi — APPROVED** [review]

File: server/routerlicious/packages/services-client/src/gitManager.ts (hunk 1 above)
Comment (appym31) [reply server/routerlicious/packages/services-client/src/gitManager.ts:11]: Sorry I missed this comment, there was a linter error that the line was too long so I had moved to multiple lines previously, and left it as is

Comment (tanviraumi) [reply server/routerlicious/packages/services-client/src/gitManager.ts:11]: It's still less than 120 right? If you create multiple lines, the convention is:

import {
    ICreateRefParamsExternal,
    IPatchRefParamsExternal,
    IGitManager,
    IHistorian
} from "./storage";

Comment (appym31) [reply server/routerlicious/packages/services-client/src/gitManager.ts:11]: Somehow missed this again, will revert it back to what it was

**Review by tanviraumi — APPROVED** [review]

**Pull request merged**